common/src/auth/ServiceConnectionStringParser.cs

```diff
@@ -0,0 +1,228 @@
+        /// <code>
+        /// HostName=[ServiceName];SharedAccessKeyName=[keyName];SharedAccessKey=[Key]
+        /// </code>
+        ///
+        /// This object parse the connection string providing the artifacts to the <see cref="ServiceConnectionString"/> object.
+        /// </remarks>
+        /// <param name="serviceConnectionString">the <code>string</code> with the connection string information.</param>
+        /// <returns>A <code>ServiceConnectionStringParser</code> object with the parsed connection string.</returns>
+        public static ServiceConnectionStringParser Create(string serviceConnectionString)
+        {
+            if (string.IsNullOrWhiteSpace(serviceConnectionString))
+            {
+                throw new ArgumentNullException(nameof(serviceConnectionString));
+            }
+
+            var ServiceConnectionStringParser = new ServiceConnectionStringParser();
+            ServiceConnectionStringParser.Parse(serviceConnectionString);
+
+            return ServiceConnectionStringParser;
+        }
+
+        /// <summary>
+        /// The Service Client Hostname
+        /// </summary>
+        public string HostName
+        {
+            get { return _hostName; }
+            set { SetHostName(value); }
+        }
+
+        /// <summary>
+        /// The Service Access Policy Name
+        /// </summary>
+        public string SharedAccessKeyName { get; internal set; }
+
+        /// <summary>
+        /// The Service Shared Access Key for the specified
+        /// access policy
+        /// </summary>
+        public string SharedAccessKey { get; internal set; }
+
+        /// <summary>
+        /// The Service Shared Access Signature
+        /// </summary>
+        public string SharedAccessSignatureString { get; internal set; }
+
+        /// <summary>
+        /// The Service Name
+        /// </summary>
+        public string ServiceName
+        {
+            get { return _serviceName; }
+        }
+
+        internal ServiceConnectionString ToServiceConnectionString()
+        {
+            Validate();
+            return new ServiceConnectionString(this);
+        }
+
+        /// <summary>
+        /// Returns the Service Connection string
+        /// </summary>
+        /// <returns></returns>
+        public override string ToString()
+        {
+            Validate();
+
+            StringBuilder stringBuilder = new StringBuilder();
+            stringBuilder.AppendKeyValuePairIfNotEmpty(HostNamePropertyName, HostName);
+            stringBuilder.AppendKeyValuePairIfNotEmpty(SharedAccessKeyNamePropertyName, SharedAccessKeyName);
+            stringBuilder.AppendKeyValuePairIfNotEmpty(SharedAccessKeyPropertyName, SharedAccessKey);
+            stringBuilder.AppendKeyValuePairIfNotEmpty(SharedAccessSignaturePropertyName, SharedAccessSignatureString);
+            if (stringBuilder.Length > 0)
+            {
+                stringBuilder.Remove(stringBuilder.Length - 1, 1);
```

Comment:
Might comment why

Reply:
Added a comment explaining we need to remove the last delimiter ";"

Reply:
Consider using TrimEnd instead then, because you can specify the character to remove, and it makes it more evident.

https://docs.microsoft.com/en-us/dotnet/api/system.string.trimend?view=netframework-4.8